cmd/thanos/query_frontend.go

```diff
@@ -82,6 +83,10 @@ func registerQueryFrontend(app *extkingpin.App) {
 	cmd.Flag("query-frontend.log-queries-longer-than", "Log queries that are slower than the specified duration. "+
 		"Set to 0 to disable. Set to < 0 to enable on all queries.").Default("0").DurationVar(&cfg.CortexFrontendConfig.LogQueriesLongerThan)
 
+	cmd.Flag("query-frontend.org-id-header", "Request header names used to set the org id field in the slow query log (repeated flag). "+
+		"If multiple headers match the request, the first matching arg specified will take precedence. "+
```

Comment:
I don't understand this bit? Can you add test for this case to make it a bit more clear?

Reply:
Thanks, yeah this sounded clearer in my head; I'll try to make the behaviour clearer in the help string.  The idea is that if `--query-frontend.org-id-header` is used multiple times, we try to extract a header from the request in order the command line args were specified.

There are test cases to check the precedence behaviour in query_frontend_test.go, e.g. we check that when both `--query-frontend.org-id-header=x-server-id` and `query-frontend.org-id-header=x-grafana-user` are specified that org id is set to the value of the x-server-id header.    However, happy to split that into a few test functions with different names if it would be clearer what's being tested?

Also, I noticed that this file has changed a bit since writing it and after rebasing it seems bit out of place.  Considering moving the new logic into pkg/queryfrontend. WDYT?

Reply:
Thanks for clarifying, now looking at the test it is clear.
I think the test location is fine, but if you prefer can move it.